### DISCUSSION TITLE
While using TasmoCompiler no I2C is shown in GPIO configuration

### DISCUSSION THREAD

**Discussion by Maddinthebrain** [arendst/Tasmota#10256] [Q&A] [answered]
Hello,

I want to use the TSL2561 and the VEML6075 sensors together as there's no precompiled bin I tried to create one using TasmoCompiler. 
The custom parameters are as below:
```
#ifndef USE_VEML6075
#define USE_VEML6075            // Add I2C code for VEML6075 sensor
#endif
#ifndef USE_TSL2561
#define USE_TSL2561         // Enable TSL2561 sensor (I2C address 0x29, 0x39 or 0x49) (+2k3 code)
#endif
```

But in Tasmotas webinterface at the GPIO Configuration there is no I2C selection, as you can see in the screenshot. 
![image](https://user-images.githubusercontent.com/43421040/103141625-2796e900-46f7-11eb-806d-35b057f2d3af.png)

Any idea?

Thx

**Comment by sfromis** [comment, marked as answer]
The screenshot does not show whether I2C configs are available or not. I'd expect to see the `I2C SDA` and `I2C SCL` options by scrolling down in the long dropdown list. I'm seeing them before a group of `SSPI` pins.

**Reply by Maddinthebrain** [reply]
Well there is no I2C SDA and I2C SCL or similar not before SSPI nor elsewere, I checked it several times. I showed the screenshot, because there are no numbers between brackets behind the entries. Maybe a hint?

Or is something missing at the custom parameters? Or some of the checkboxes must be selected? I guess I2C is normally included, isn't it?

**Reply by sfromis** [reply]
What I reacted to was that your screenshot in no way shows if I2C gpio settings are available. Using [decode-status](https://github.com/tasmota/Tasmota-decode-status) you can check what features (like `USE_I2C`) are included in your binary. I'd generally expect it to be, but better check.

You did not mention your concern about no bracketed numbers behind entries, this is merely a change with version 9 having a new GPIO numbering system, and the dropdown shortened by having an extra dropdown to the right for some types of pins. This made it unfeasible to have numbers included in the dropdown.

**Comment by Jason2866** [comment]
Post the output of `gpio all`

**Reply by Frogmore42** [reply]
What did you use as the base for TasmoCompile?
Have you tried a pre-compiled version?  I2C is in the default version but not in the minimal version.

Either the version you compiled has i2c support or it doesn't.  If it doesn't, the next question is why doesn't it?  I don't use TasmoCompile, but it looks pretty simple.  It also looks like it hides a lot of the complexity from you, so when things go wrong it is harder to figure out.  You could just add the defines for USE_I2C.  But, try the standard binary and see if you see the I2C option.

**Comment by Maddinthebrain** [comment]
I added #define USE_I2C to custom parameters. Now I2C SDA and I2C SCL appeared... Can somebody check this?

**Reply by Frogmore42** [reply]
This is more likely a discussion for TasmoCompile.  Since it works when you specify to use i2c, Tasmota is working as expected.  My guess is you are not using TasmoCompile correctly, or your expectation for it is not what it does.

The base configuration of Tasmota (minimal) does not include i2c.  You want i2c, so you have to specify that.  But, your question is really all about TasmoCompile and how it works, not Tasmota.

**Reply by Maddinthebrain** [reply]
Well that's the point. I don't really know which variant in TasmoCompiler is used, it's not written there. TasmoCompiler is convenient to use, maybe a bit too convenient...

Thank you for your support!

**Reply by Frogmore42** [reply]
Look here:
https://github.com/benzino77/tasmocompiler

There is a check box for i2c sensors.  That is probably something you missed checking.  The fact that the checkbox is in the UI, indicates it is probably using minimal, or even less, for the default configuration.

The platformio.ini file would probably also indicate that i2c was not selected.  That is the problem with things that make it easy.  They are great when they just work.  But, when they don't, they can make it even harder to figure out what you need to do to fix it.

But, it looks like your problem is solved.  You shouldn't need to use the #define for i2c, but you do need to select it in the GUI.

**Reply by benzino77** [reply]
I've removed I2C and SPI options from the GUI to not confuse inexperienced users. Those features are included in the "background" if you select a feature or feature group which requires it. There is also a file which describes what features are included if you select given checkbox: [FEATURES_DESC](https://github.com/benzino77/tasmocompiler/blob/master/FEATURES_DESC.md) - maybe there is a need to put link to this document somewhere in the GUI...

I do believe that TC needs new features group called "Light sensors" which will include TSL2561 and the VEML6075 (and probably others - suggestions are welcome).

If you are concern what TC is doing under the hood just download `user_config_override.h` file to see which `#defines` are inside. Current version of TC is using it's own structure of platformio files, but in a day or two I will release RC version of TC which will be using only `user_config_override.h` and `platformio_override.ini` files. It will also add full support for ESP32 boards.

**Reply by benzino77** [reply]
https://github.com/benzino77/tasmocompiler/discussions/121
